src/lib/utils/options/readers/package-json.ts

```diff
@@ -0,0 +1,84 @@
+import { dirname, join, resolve } from "path";
+import * as fs from "fs";
+
+import type { OptionsReader } from "..";
+import type { Logger } from "../../loggers";
+import type { Options } from "../options";
+import { ok } from "assert";
+import { nicePath } from "../../paths";
+
+export class PackageJsonReader implements OptionsReader {
+    // Should run after the TypeDoc config reader but before the TS config
+    // reader, so that it can still specify a path to a `tsconfig.json` file.
+    priority = 150;
+
+    name = "package-json";
+
+    read(container: Options, logger: Logger): void {
+        const cfgPath = container.getValue("options");
+        const absPath = resolve(cfgPath);
+
+        // Search for `package.json` first at the provided config path, or at
+        // `cfgPath/package.json` if the provided path is a directory.
+        const path = [absPath, join(absPath, "package.json")].find(
+            (p) => fs.existsSync(p) && fs.statSync(p).isFile()
+        );
+
+        if (!path) {
+            if (container.isSet("options")) {
+                logger.error(
+                    `A package.json file at ${nicePath(
+                        cfgPath
+                    )} does not exist.`
+                );
+            }
+            return;
+        }
+
+        const content = fs.readFileSync(path, "utf-8");
+
+        let pkg: Record<string, any>;
+        try {
+            pkg = JSON.parse(content);
+            if (pkg === null || typeof pkg !== "object") {
+                this.logParsePkgFailure(logger, path);
+                return;
+            }
+        } catch (err) {
+            ok(err instanceof Error);
+            this.logParsePkgFailure(logger, path);
+            return;
+        }
+
+        const opts: Record<string, any> | undefined = pkg["typedocOptions"];
+        if (opts === null || typeof opts !== "object") {
```

Comment:
It should not be an error if this key is not set, in that case, the user is likely configuring typedoc through another method.

Reply:
Okay sounds good. We'll first check if the key exists or not, and only raise an error if the key is set but the data is invalid.